lib/utils/setupExitSignals.js

```diff
@@ -1,21 +0,0 @@
-'use strict';
-
-const signals = ['SIGINT', 'SIGTERM'];
-
-function setupExitSignals(serverData) {
-  signals.forEach((signal) => {
-    process.on(signal, () => {
-      if (serverData && serverData.server) {
-        serverData.server.close(() => {
-          // eslint-disable-next-line no-process-exit
-          process.exit();
-        });
-      } else {
-        // eslint-disable-next-line no-process-exit
-        process.exit();
-      }
-    });
-  });
-}
-
-module.exports = setupExitSignals;
```

Comment:
We should keep it, it is very important, otherwise `CTRL+C` will be broken, maybe put it here https://github.com/webpack/webpack-dev-server/blob/master/lib/Server.js#L777?

Reply:
Shouldn't this be part of the CLI, `@webpack-cli/serve`?

Reply:
hm, I think no, webpack-cli should don't know about internal logic + when we will refactor it to plugin it will be impossible, we call https://github.com/webpack/webpack-cli/blob/master/packages/serve/src/startDevServer.ts#L48 and we can put logic in `listen`, in think logic will work only when `options.port` is undefined

I will fix problem here https://github.com/webpack/webpack-cli/blob/master/packages/serve/src/startDevServer.ts#L36, after we do release v4

Reply:
Wrong window please :smile: No, it is internal, we should keep internal logic for server inside our code

Reply:
webpack-cli already do many logic right now, in the future it should only put only the plugin and nothing more

Reply:
This behavior might be undesirable when called directly via the API. Maybe add an option, `setupExitSignal`?

Reply:
Yep, source good for the new option for API